src/WebJobs.Extensions.DurableTask/ContextInterfaces/IDurableOrchestrationContext.cs

```diff
@@ -339,5 +340,13 @@ public interface IDurableOrchestrationContext
         /// </exception>
         /// <returns>The instance id of the new orchestration.</returns>
         string StartNewOrchestration(string functionName, object input, string instanceId = null);
+
+        /// <summary>
+        /// Returns an instance of ILogger that is replay safe, ensuring the logger logs only when the orchestrator
+        /// is not replaying that line of code.
+        /// </summary>
+        /// <param name="logger">An instance of ILogger.</param>
+        /// <returns>An instance of a replay safe ILogger.</returns>
+        ILogger CreateReplaySafeLogger(ILogger logger);
```

Comment:
Instead of adding this to the interface and implementing it directly on `DurableOrchestrationContext`, let's make it an extension method. You can put it in `DurableContextExtensions`. This way, the customer gets access to our implementation of `CreateReplaySafeLogger()` even if they aren't using the `DurableOrchestrationContext` implementation. Otherwise they will need to mock that behavior themselves.